subgraph/src/mappings/request-manager.ts

```diff
@@ -0,0 +1,147 @@
+  const request = Request.load(requestId);
+
+  if(claim && request && claim.resolved == false) {
+    claim.resolved = true;
+    request.activeClaims = request.activeClaims.minus(BigInt.fromI32(1));
+    claim.save();
+    request.save();
+  }
+}
+export function handleDepositWithdrawn(event: DepositWithdrawn): void {
+  const requestId = event.params.requestId;
+  const receiver = event.params.receiver;
+  const request = Request.load(requestId);
+
+  if (request) {
+   // If the receiver is the request sender, then the user is withdrawing
+    // because the request has expired with no fill
+    if(request.sender.toHexString() == receiver.toHexString()) {
+      request.withdrawClaimId = (CLAIM_ID_EXPIRED_REQUEST);
+    } else {
+      const claims = request.claims;
+      // iterate over claims and find the one with claimer event.params.receiver
+      // if found, set the withdrawClaimId to that claim.id
+      for (let i = 0; i < claims.length; i++) {
+        const claim = Claim.load(claims[i]);
+        if (claim && claim.claimer == event.params.receiver) {
+          request.withdrawClaimId = BigInt.fromString(claim.id);
+          break;
+        }
+      }
+    }
+
+    request.save();
+  }
+}
+
+export function handleRequestCreated(event: RequestCreated): void {
+  let request = new Request(event.params.requestId);
+  request.sender = event.params.sourceAddress;
+  request.sourceTokenAddress = event.params.sourceTokenAddress;
+  request.targetTokenAddress = event.params.targetTokenAddress;
+  request.targetChainId = event.params.targetChainId;
+  request.amount = event.params.amount;
+  request.validUntil = event.params.validUntil;
+  request.lpFee = event.params.lpFee;
+  request.protocolFee = event.params.protocolFee;
+  request.activeClaims = ZERO_BI;
+  request.withdrawClaimId = ZERO_BI;
+  request.filler = ADDRESS_ZERO_BYTES;
+  request.fillId = ZERO_BYTES;
+  request.targetAddress = event.params.targetAddress;
+  request.nonce = event.params.nonce;
+  request.invalidFills = [];
+  request.claims = [];
+  request.transaction = loadTransaction(event).id
+  request.emittedAt = event.block.timestamp
+  request.emittedByTx = event.transaction.hash
+  request.contractAddress = event.address
+
+  request.save();
+}
+
+export function handleFillInvalidatedResolved(event: FillInvalidatedResolved): void {
+  const request = Request.load(event.params.requestId);
+
+  if (request) {
+    const invalidFills = request.invalidFills || [];
+
+    for (let i = 0; i < invalidFills.length; i++) {
+      const fillState = FillState.load(invalidFills[i]);
+      if (fillState && fillState.id == event.params.fillId) {
+        fillState.invalid = true;
+        fillState.save();
+        break;
+      }
+    }
+  }
+}
+
+export function handleRequestResolved(event: RequestResolved): void {
+  const request = Request.load(event.params.requestId);
+
+  if (request) {
+    const invalidFills = request.invalidFills ? request.invalidFills : [];
```

Comment:
uff, those fancy developers, with their fancy new syntax :D 

Nullish coalescing operator was added to javascript in about 2020, you are talking with someone who grew up in 
`var self = this`, prototype calls and other stuff you basically never touch those days. :D

Reply:
Haha, i knew exactly what your response would be :P Some fresh syntax for you :D

>BTW - I'm not sure how long I'll have to use ?? to get used to it to immediately understand what it is supposed to mean. got so used to ? : ...

Until the moment you start getting used to it, it will probably be replaced by something else xD